src/actions/recomputeReduxState.js

```diff
@@ -696,8 +697,9 @@ export const createStateFromQueryOrJSONs = ({
     controls["absoluteZoomMin"] = 0;
     controls["absoluteZoomMax"] = entropy.lengthSequence;
   } else if (oldState) {
-    /* revisit this - but it helps prevent bugs */
-    controls = {...oldState.controls};
+    /* creating deep copies avoids references to (nested) objects remaining the same which
+    can affect props comparisons. Due to the size of some of the state, we only do this selectively */
```

Comment:
This makes sense and as far as I can tell aligns with [the redux FAQ](https://redux.js.org/faq/performance#do-i-have-to-deep-clone-my-state-in-a-reducer-isnt-copying-my-state-going-to-be-slow), namely 

> Common Redux misconception: you need to deeply clone the state. Reality: if something inside doesn't change, keep its reference the same!

in that here it seems like we are abiding by that in general with shallow copies and specifically deep cloning `controls` due to the coincidence of calls to `createStateFromQueryOrJSONs` (with `oldState`) with changes to the controls state that will require props comparison. That is as opposed to some of the other slices of the state which are more tied to dataset and so don't rely on as many comparisons since dataset changes mostly come with a fresh redux state.